src/content/blog/ga_qas.md

```diff
@@ -0,0 +1,233 @@
+---
+title: An Introduction to GA-QAS
+author: Vu Tuan Hai
+day: 21
+month: 5
+year: 2024
+tags:
+  - quantum architecture search
+  - python
+  - qiskit
+  - genetic-algorithm
+---
+
+In various quantum optimization problems, choosing the right ansatz is a critical point that will affect the accuracy of the result. Many template ansatzes have been proposed, such as Graph Ansatz and EffecientSU2, … but they are limited in their applicability. Hence, we've created a search engine called **GA-QAS** (Genetic Algorithm for Quantum Architecture Search) to aid in the discovery of the right ansatz. In this post, we will guide you on how to use it efficiently.
+
+Required packages:
+
+- qiskit 1.0+
+- matplotlib
+- tdqm
+
+## Now, let’s start!
+
+# Step 1. Define problem (W state preparation)
+
+<figure>
+    <img src='/public/images/ga-qas/ga-qas1.png' width="600" style="margin: auto" alt='description' />
+    <figcaption>
+    Fig 1. (a) State preparation scheme based on quantum compilation technique, where U, V are ansatz and state that need to prepare[1][2]. (b) Some default ansatz, such as hyper-graph.
+    </figcaption>
+</figure>
+
+In this post, we will use a Python package named <a href="https://github.com/vutuanhai237/qoop">$\langle \mathrm{qo}|\mathrm{op} \rangle$</a>, which is a core package for developing quantum circuit optimization frameworks.
+You can download it via `git` as:
+
+```
+git clone https://github.com/vutuanhai237/qoop.git
+```
+
+Note that you should put it on the same level as your Python/Jupyter Notebook file. The best is:
+
+```
+folder_name
+│   your_python_file.py
+|   your_jupyter_file.ipynb
+└───qoop
+│   └───evolution
+│       │   environment.py
+│       │   cross_over.py
+│       │   ...
+```
+
+First, we define the problem, take example quantum state preparation, more easier W state preparation. 3-qubit W state is defined as:
```

Comment:
I think we want to say something along the lines of

> One of the core problems in quantum computing is that of state preparation. In this demonstration we will use `qoop` to demonstrate how to find a unitary $V$ that creates the 3-qubit Werner state defined below.

Reply:
@vutuanhai237 don't forget this suggestion! The current sentence is a bit confusing.

Reply:
@vutuanhai237 this bit needs to be reworked to bring in readers who need more context as to the usefulness of your package. I made a suggestion above. Can you ensure there's a nice entry point for readers who are less familiar with this specific problem.

Reply:
@vutuanhai237 reminder that this comment needs to be addressed in order to move things forward.

Reply:
Sorry @natestemen , look like I missed some above comments!

Reply:
I resolved it in the above.